web-marketplace/src/pages/BuyCredits/BuyCredits.Form.tsx

```diff
@@ -459,6 +517,41 @@ export const BuyCreditsForm = ({
         modalState={modalState}
         redirectRoute={`${projectHref.replace(/^\//, '')}/buy`}
       />
+      {fiatModalContent.current && (
+        <BuyFiatModal
+          modalContent={fiatModalContent.current.modalContent}
+          fiatModalState={fiatModalState}
+          onClose={setFiatModalState}
+          handleClick={action => {
+            if (action === KEPLR_LOGIN_REQUIRED) {
+              setSwitchWalletModalAtom(atom => void (atom.open = true));
```

Comment:
see existing logic in `BuyCredits.Form.tsx` (`setupWalletModal` prop), in certain cases we need to display ConnectWalletModal:

```
if (!activeWalletAddr) {
  // no connected wallet address
  setConnectWalletModal(atom => void (atom.open = true));
} else if (!isConnected) {
  // user logged in with web2 but not connected to the wallet address associated to his/er account
  setSwitchWalletModalAtom(atom => void (atom.open = true));
}
```

Reply:
@r41ph could you have a look at this?